### PR TITLE
Printing resource labels as columns

### LINKED ISSUE #9631: Proposal for extending kubectl get [resource] by printing resource labels as extra columns
### **Proposal**

After finishing with #9032, we can also add resource labels as extra columns when we print resources. 

The user could define in a flag `--label-column=` or `-l` the names of the labels that he wants to print, and then **kubectl will append extra columns in the end of each resource row**. 

The extra column will use:
* as header-column name --> the label name
* as per row value --> the label value for each resource

For labels:
* names can be both in lowercase or uppercase
* non-recognized labels will return `<n/a>` to explicitly declare that this label is not available in this resource _(as labels are strings, we can differentiate empty string "" with non-existing label)_ **???**
* declaration order on the command reflects on printing

### **Example**
```
$ kubectl get pods --label-column=version,deployment
NAME             READY     REASON    RESTARTS   AGE  VERSION  DEPLOYMENT
my-nginx-00jsd   0/1       Pending   0          ~1m  v1       2016-01-01
my-nginx-7kqqp   0/1       Pending   0          ~1m  v1       2016-01-01
my-nginx-fb2s6   0/1       Pending   0          ~1m  v2       2016-01-02
```

/cc @ghodss 

*I will update this comment during the conversation with the latest scheme we have agreed on. This first proposal is just to initialise a conversation.*

### PR DESCRIPTION
**Pull request by andronat**
Fixes #9631

By adding `--label-column=` you can print any label as column.

Question: Are all labels capital? Do we to be case-sensitive?

### CODE REVIEW

File: contrib/completions/bash/kubectl [hunk 1]
```diff
@@ -232,6 +232,8 @@ _kubectl_get()
     flags+=("--all-namespaces")
     flags+=("--help")
     flags+=("-h")
+    flags+=("--label-columns=")
+    two_word_flags+=("-k")
```
Comment (bgrant0607) [inline contrib/completions/bash/kubectl]: `-L` is a good idea. Let's go with that.

**Comment by bgrant0607** [comment]
Looks good, though let's switch to `-L`.

The test failure looks like a flake.

**Comment by andronat** [comment]
Done, rebased and tests are working on my laptop. Should I squeeze the commits?

File: docs/kubectl_get.md [hunk 2]
```diff
@@ -46,6 +46,7 @@ $ kubectl get rc/web service/frontend pods/web-pod-13je7
 ```
       --all-namespaces=false: If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.
   -h, --help=false: help for get
+  -L, --label-columns=[]: Accepts a comma separated list of labels that are going to be presented as columns. Names are case-sensitive. You can also use multiple flag statements like -k label1 -k label2...
```
Comment (bgrant0607) [inline docs/kubectl_get.md]: The description still mentions `-k`

**Comment by bgrant0607** [comment]
Yes, go ahead and squash after fixing the -k issue. The Shippable failure looks like a flake.

File: docs/kubectl_get.md (hunk 2 above)
Comment (andronat) [inline docs/kubectl_get.md]: Sorry! Fixed

**Comment by bgrant0607** [comment]
ok to test

**Comment by andronat** [comment]
Squashed and ready

**Comment by bgrant0607** [comment]
Thanks. LGTM, assuming tests pass.

**Comment by andronat** [comment]
I am not sure why this fails

**Comment by bgrant0607** [comment]
Shippable passed. Looking at e2e output, but I doubt it's caused by this PR.

**Comment by bgrant0607** [comment]
The failure was the addon_upgrade test recently added by PR #9493. It's not invoking the functionality you're adding, so I think it's a flake.

**Comment by jlowdermilk** [comment]
@k8s-bot retest this please

**Comment by jlowdermilk** [comment]
@k8s-bot retest this please

**Comment by ixdy** [comment]
Due to a bug in the PR plugin, the bot failed to update, but the most recent run succeeded.

Ran 48 of 89 Specs in 322.395 seconds
SUCCESS! -- 48 Passed | 0 Failed | 2 Pending | 39 Skipped 

GCE e2e build/test **passed** for commit b06ef70057ee8195d82a9fdc132dcd88b596f2f3.
* [Build Log](https://storage.cloud.google.com/kubernetes-jenkins/pr-logs/b06ef70057ee8195d82a9fdc132dcd88b596f2f3/kubernetes-pull-build-test-e2e-gce/794/build-log.txt)
* [Test Artifacts](https://console.developers.google.com/storage/browser/kubernetes-jenkins/pr-logs/b06ef70057ee8195d82a9fdc132dcd88b596f2f3/kubernetes-pull-build-test-e2e-gce/794/_artifacts/)
* [Internal Jenkins Results](http://goto.google.com/prkubekins/job/kubernetes-pull-build-test-e2e-gce//794)

**Pull request merged**